R/unnecessary_nesting_linter.R

```diff
@@ -14,8 +14,12 @@
 #'   testthat requires a braced expression in its `code` argument. The other defaults
 #'   similarly compute on expressions in a way which is worth highlighting by
 #'   em-bracing them, even if there is only one expression, while [switch()] is allowed
-#'   for its use as a control flow analogous to `if`/`else`.
-#'
+#'   for its use as a control flow analogous to `if`/`else`.]
+#' @param branch_exit_calls Character vector of functions which are considered
+#'   as "exiting" a branch for the purpose of recommending removing nesting in a branch
+#'   _lacking_ an exit call when the other branch terminates with one. Calls which
+#'   always interrupt or quit the current call or R session,
+#'   e.g. [stop()] and [q()], are always included.
```

Comment:
It might be a good idea to include all elements of the relevant vector (`default_branch_exit_calls`) in the docs. 

Not everyone is going to be able to recall all the functions that interrupt or quit the current call or R session.

Reply:
I am a bit wary of this. It's a bit fragile / easy to get out of sync with the true set of calls, and extremely easy to figure out what the defaults are by glancing at the code.

I will treat this as an implementation detail & wait for a user request here before proceeding.